prereise/gather/flexibilitydata/doe/doe_data.py

```diff
@@ -0,0 +1,112 @@
+
+    all_folders = os.listdir(root)
+    for i in all_folders:
+
+        if os.path.isdir(os.path.join(root, i)):
+            cleanup_directory(os.path.join(root, i))
+
+    return
+
+
+def download_doe(download_path="data"):
+    """Download demand flexibility filters from OEDI, extract and cleanup
+
+    :param str download_path: the directory where the original DOE data will be stored
+    """
+
+    # create data directory
+    os.makedirs(download_path, exist_ok=True)
+
+    # download zip data
+    oedi_filter_link = (
+        "https://data.openei.org/files/180/2006weatherentireusdrfilters.tar.zip"
+    )
+    urllib.request.urlretrieve(
+        oedi_filter_link, os.path.join(download_path, "filter.zip")
+    )
+
+    # extract
+    with zipfile.ZipFile(os.path.join(download_path, "filter.zip"), "r") as fh:
+        fh.extractall(download_path)
+
+    # delete and further extract
+    os.remove(os.path.join(download_path, "filter.zip"))
+
+    for i in os.listdir(download_path):
+        fh = tarfile.open(os.path.join(download_path, i))
+        fh.extractall(download_path)
+
+        fh.close()
+        os.remove(os.path.join(download_path, i))
+
+    # cleanup
+    cleanup_directory(download_path)
+
+
+def aggregate_doe(root, out_path):
+    """Aggregate sector flexibilties by summing up the percentage flexibility from all sectors
+    and store to output csv file
+
+    :param str root: the root directory containing raw de-compressed DOE flexibility data
+    :param str out_path: the output file where the aggregated data will be stored
+    """
+
+    all_folders = os.listdir(root)
+
+    # initialize output container
+    eia_flex = pd.DataFrame(np.zeros((8808, len(all_folders))), columns=all_folders)
+
+    for i in all_folders:
+        all_csvs = os.listdir(os.path.join(root, i))
+
+        # new column for total flexibility
+        file_flex = pd.read_csv(os.path.join(root, i, all_csvs[0]), index_col=0)
+
+        # assign index to the output df
+        if "time" not in eia_flex.keys():
+            eia_flex["time"] = pd.to_datetime(file_flex.index.copy())
+            eia_flex.set_index("time", inplace=True)
+
+        eia_flex[i] = eia_flex[i] + file_flex["Flexibility"].values
+        eia_flex.at[eia_flex.index == file_flex.index, i] = file_flex[
+            "Flexibility"
+        ].values
+
+        for c in all_csvs[1:]:
+            fn = os.path.join(root, i, c)
+            file_flex = pd.read_csv(fn, index_col=0)
+
+            if file_flex.shape[0] == eia_flex.shape[0]:
+                eia_flex[i] = eia_flex[i] + file_flex["Flexibility"].values
+            else:
+                eia_flex.loc[
+                    eia_flex.index.isin(pd.to_datetime(file_flex.index)), i
+                ] = (
+                    eia_flex.loc[
+                        eia_flex.index.isin(pd.to_datetime(file_flex.index)), i
+                    ]
+                    + file_flex["Flexibility"].values
+                )
```

Comment:
I just read the doc, it seems update replaces the column but not add them up?

Reply:
Ah, I didn't realize we are doing the addition here, then this might work
```python
eia_flex[i] = eia_flex[i].add(file_flex["Flexibility"], fill_value=0)
```

Reply:
let me try it